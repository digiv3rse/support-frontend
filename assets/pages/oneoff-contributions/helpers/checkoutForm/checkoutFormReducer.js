// @flow

// ----- Imports ----- //
import { type Action } from './checkoutFormActions';

// ----- Types ----- //


export type CheckoutFormAttribute = {
  shouldValidate: boolean,
}

export type OneOffContributionsCheckoutFormState = {
  email: CheckoutFormAttribute,
  fullName: CheckoutFormAttribute,
};

// ----- Setup ----- //

const initialState: OneOffContributionsCheckoutFormState = {
  email: {
    shouldValidate: false,
  },
  fullName: {
    shouldValidate: false,
  },
};


// ----- Reducer ----- //

function checkoutFormReducer(
  state: OneOffContributionsCheckoutFormState = initialState,
  action: Action,
): OneOffContributionsCheckoutFormState {

  switch (action.type) {
    case 'SET_EMAIL_SHOULD_VALIDATE':
      return { ...state, email: { ...state.email, shouldValidate: true } };

    case 'SET_FULL_NAME_SHOULD_VALIDATE':
      return { ...state, fullName: { ...state.fullName, shouldValidate: true } };

    default:
      return state;
  }

}

export { checkoutFormReducer };
