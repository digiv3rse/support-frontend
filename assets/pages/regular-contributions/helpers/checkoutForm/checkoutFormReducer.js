// @flow

// ----- Imports ----- //
import { type Action } from './checkoutFormActions';

// ----- Types ----- //


export type CheckoutFormAttribute = {
  shouldValidate: boolean;
}

export type Stage = 'checkout' | 'payment'

export type RegularContributionsCheckoutFormState = {
  email: CheckoutFormAttribute,
  firstName: CheckoutFormAttribute,
  lastName: CheckoutFormAttribute,
  stage: Stage,
};

// ----- Setup ----- //

const initialState: RegularContributionsCheckoutFormState = {
  email: {
    shouldValidate: false,
  },
  firstName: {
    shouldValidate: false,
  },
  lastName: {
    shouldValidate: false,
  },
  stage: 'checkout',
};


// ----- Reducer ----- //

function checkoutFormReducer(
  state: RegularContributionsCheckoutFormState = initialState,
  action: Action,
): RegularContributionsCheckoutFormState {

  switch (action.type) {
    case 'SET_FIRST_NAME_SHOULD_VALIDATE':
      return { ...state, firstName: { ...state.firstName, shouldValidate: action.shouldValidate } };

    case 'SET_LAST_NAME_SHOULD_VALIDATE':
      return { ...state, lastName: { ...state.lastName, shouldValidate: action.shouldValidate } };

    case 'SET_EMAIL_SHOULD_VALIDATE':
      return { ...state, email: { ...state.email, shouldValidate: action.shouldValidate } };

    case 'SET_STAGE':
      return { ...state, stage: action.stage };

    default:
      return state;
  }

}

export { checkoutFormReducer };
