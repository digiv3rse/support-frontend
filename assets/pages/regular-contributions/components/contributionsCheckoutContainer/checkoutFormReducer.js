// @flow

// ----- Imports ----- //
import { emailRegexPattern } from 'helpers/checkoutForm/checkoutForm';
import { type Action } from './checkoutFormActions';

// ----- Types ----- //


export type CheckoutFormAttribute = {
  shouldValidate: boolean,
  required: boolean,
  pattern: RegExp,
}

export type RegularContributionsCheckoutFormState = {
  email: CheckoutFormAttribute,
  firstName: CheckoutFormAttribute,
  lastName: CheckoutFormAttribute,
};

// ----- Setup ----- //

const initialState: RegularContributionsCheckoutFormState = {
  email: {
    shouldValidate: false,
    required: true,
    pattern: emailRegexPattern,
  },
  firstName: {
    shouldValidate: false,
    required: true,
    pattern: /.*/,
  },
  lastName: {
    shouldValidate: false,
    required: true,
    pattern: /.*/,
  },
};


// ----- Reducer ----- //

function checkoutFormReducer(
  state: RegularContributionsCheckoutFormState = initialState,
  action: Action,
): RegularContributionsCheckoutFormState {

  switch (action.type) {
    case 'SET_FIRST_NAME_SHOULD_VALIDATE':
      return { ...state, firstName: { ...state.firstName, shouldValidate: true } };

    case 'SET_LAST_NAME_SHOULD_VALIDATE':
      return { ...state, lastName: { ...state.lastName, shouldValidate: true } };

    case 'SET_EMAIL_SHOULD_VALIDATE':
      return { ...state, email: { ...state.email, shouldValidate: true } };

    default:
      return state;
  }

}

export { checkoutFormReducer };
