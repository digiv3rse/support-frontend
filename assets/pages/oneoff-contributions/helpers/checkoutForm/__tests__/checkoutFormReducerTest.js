import { checkoutFormReducer as reducer } from '../checkoutFormReducer';

// ----- Tests ----- //

describe('checkout form reducer tests', () => {

  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle SET_EMAIL_SHOULD_VALIDATE', () => {
    const action = { type: 'SET_EMAIL_SHOULD_VALIDATE', shouldValidate: true };

    const newState = reducer(undefined, action);

    expect(newState.email.shouldValidate).toEqual(true);
  });

  it('should handle SET_FULL_NAME_SHOULD_VALIDATE', () => {
    const action = { type: 'SET_FULL_NAME_SHOULD_VALIDATE', shouldValidate: true };

    const newState = reducer(undefined, action);

    expect(newState.fullName.shouldValidate).toEqual(true);
  });

});
