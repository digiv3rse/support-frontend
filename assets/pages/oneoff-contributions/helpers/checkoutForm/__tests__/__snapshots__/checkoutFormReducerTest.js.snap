// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkout form reducer tests should return the initial state 1`] = `
Object {
  "email": Object {
    "shouldValidate": false,
  },
  "fullName": Object {
    "shouldValidate": false,
  },
}
`;
